microsetta_admin/templates/add_barcode_to_kit.html

```diff
@@ -0,0 +1,160 @@
+                        // Handle success response
+                        console.log(response);
+                    },
+                    error: function(xhr, status, error) {
+                        // Handle error response
+                        console.error(error);
+                    }
+                });
+            }
+        });
+
+        // Track which button was clicked
+        $("input[type='submit']").click(function() {
+            $("input[type='submit']", $(this).parents("form")).removeAttr("clicked");
+            $(this).attr("clicked", "true");
+        });
+    });
+
+    function setBarcodeFromList() {
+        var barcodeList = document.getElementById("barcodeList");
+        if (barcodeList && barcodeList.children.length > 0) {
+            var barcode = barcodeList.children[0].innerText;
+            document.getElementById("user_barcode").value = barcode;
+        }
+        else {
+            alert("No barcodes to generate");
+        }
+    }
+
+</script>
+
+{% endblock %}
+{% block content %}
+<h3>Add Barcode to Kit(s)</h3>
+<style>
+    select {
+        width: 250px;
+        margin: 10px;
+    }
+</style>
+<div style="height: 335px;">
+    {% if error_message %}
+        <p style="color:red">
+            {{ error_message |e }}
+        </p>
+    {% endif %}
+
+    <form name="kit_form" id="kit_form" method="POST" enctype="multipart/form-data">
+        <table>
+            <tr>
+                <td><label for="project_id">Project ID: </label></td>
+                <td><input type="number" name="project_id" id="project_id"></td>
+            </tr>
+            <tr>
+                <td><label for="num_kits">Number of kit(s): </label></td>
+                <td><input type="number" name="num_kits" id="num_kits" min="1"></td>
+            </tr>
+            <tr>
+                <td><label for="num_samples">Number of sample(s) per kit: </label></td>
+                <td><input type="number" name="num_samples" id="num_samples" min="1"></td>
+            </tr>
+            <tr>
+                <td><label for="kit_id">Kit ID: </label></td>
+                <td><input type="string" name="kit_id" id="kit_id"></td>
+            </tr>
+            <tr>
+                <td><label for="upload_csv">Or upload a CSV file of Kit IDs and/or Barcodes:</label></td>
+                <td><input type="file" name="upload_csv" id="upload_csv"></td>
+            </tr>
+            <tr>
+                <td></td>
+                <td><input type="submit" name="upload" value="Upload CSV"></td>
+            </tr>
+            <tr>
+                
+                <td><label for="user_barcode">Enter barcode: </label></td>
+                <td>
+                    <input 
+                        type="text" 
+                        name="user_barcode" 
+                        id="user_barcode" 
+                        title="Barcode must start with 'X' followed by 8 digits" 
+                        maxlength="9"
```

Comment:
```suggestion
```
These constraints only exist on the barcodes our system generates - we do not want to enforce any constraints on the barcodes a user adds based on text input/CSV upload, other than to ensure they don't already exist in our system (which should be at the API level).